zapcore/console_encoder.go

```diff
@@ -64,8 +63,8 @@ func (c consoleEncoder) Clone() Encoder {
 	return consoleEncoder{c.jsonEncoder.Clone().(*jsonEncoder)}
 }
 
-func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) ([]byte, error) {
-	line := bytes.NewBuffer(buffers.Get())
+func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) (*buffer.Buffer, error) {
```

Comment:
I'm not very against it -- I do like the separation between encoders and writers, just wanted to make sure we were getting some other advantages for the cost of another public type.